### PR TITLE
Add change endpoint name command -XDR

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
* Added get prevalence to PB

* Added get prevalence to PB

* Added get prevalence to PB

* Added get prevalence to PB

* Added get prevalence to PB

* Added get prevalence to PB

* Updated release notes

* fixed validation error

* Added DBotScore to verdict conditional task

* Added DBotScore to verdict conditional task

* Update playbook-AWS_IAM_User_Access_Investigation.yml

* updated get prevalence with dbot score

* updated release notes

* rollback

* Update QRadar_v3-test.yml

* Update playbook-SMB_test.yml

* Update 1_3_16.md

* updated release notes

* fixed conditional task logic

* removed changes

* removed changes

* Update 1_3_24.md

* removed prevalence check from PB

* remove prevalence check from playbook

* moved prevalence check to non malicious path

* updated release notes

* fixed comment and updated PB image

* updated release notes

* changed to original command instead of decoded

* removed changes for ransomware PB

* removed changes for rnasomware PB

* fixed validations

* updated release notes

* updated release notes

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-612

## Description
Add a command to XDR-IR  integration, that allows the user to change the name of the endpoint.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 5.0.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Core/ReleaseNotes/1_4_0.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Investigation & Response
+- Added A new command.
```
Comment (ShirleyDenkberg) [inline Packs/Core/ReleaseNotes/1_4_0.md:4]:
```suggestion
Added the ***xdr-endpoint-alias-change*** command that gets a list of endpoints according to the passed filters, and changes their alias name.
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 2]
```diff
@@ -3,3183 +3,3281 @@ commonfields:
...
+          predefined:
+            - none
+            - pending
+            - in_progress
+            - canceled
+            - aborted
+            - pending_cancellation
+            - success
+            - error
+      description: Gets a list of endpoints according to the passed filters, and changes their alias name. Filtering by multiple fields will be concatenated using AND condition (OR is not supported).
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:1019]:
```suggestion
      description: Gets a list of endpoints according to the passed filters, and changes their alias name. Filtering by multiple fields will be concatenated using the AND condition (OR is not supported).
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 3]
```diff
@@ -3,3183 +3,3281 @@ commonfields:
...
+            Example: dist_name1,dist_name2
+          isArray: true
+          name: dist_name
+        - description: |-
+            A comma-separated list of IP addresses.
+            Example: 8.8.8.8,1.1.1.1
+          isArray: true
+          name: ip_list
+        - description: |-
+            The group name to which the agent belongs.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:944]:
```suggestion
            A comma-separated list of group names to which the agent belongs.
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 4]
```diff
@@ -3,3183 +3,3281 @@ commonfields:
...
+            Example: group_name1,group_name2
+          isArray: true
+          name: group_name
+        - description: 'The endpoint platform. Valid values are\: "windows", "linux", "macos", or "android". '
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:948]:
```suggestion
        - description: 'A comma-separated list of endpoint platforms. Valid values are\: "windows", "linux", "macos", or "android". '
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 5]
```diff
@@ -3,3183 +3,3281 @@ commonfields:
...
+          isArray: true
+          name: alias_name
+        - auto: PREDEFINED
+          description: Specifies whether the endpoint was isolated or unisolated.
+          name: isolate
+          predefined:
+            - isolated
+            - unisolated
+        - description: |-
+            Hostname
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:969]:
```suggestion
            A comma-separated list of hostnames.
```

File: Packs/CortexXDR/TestPlaybooks/Test_XDR_Playbook_general_commands.yml [hunk 6]
```diff
@@ -2204,4 +2631,3 @@ view: |-
   }
 inputs: []
 outputs: []
-fromversion: 5.0.0
```
Comment (JasBeilin) [inline Packs/CortexXDR/TestPlaybooks/Test_XDR_Playbook_general_commands.yml:2207]: Why was this removed?

File: Packs/CortexXDR/pack_metadata.json [hunk 7]
```diff
@@ -2,7 +2,7 @@
     "name": "Cortex XDR by Palo Alto Networks",
     "description": "Automates Cortex XDR incident response, and includes custom Cortex XDR incident views and layouts to aid analyst investigations.",
     "support": "xsoar",
-    "currentVersion": "4.10.13",
+    "currentVersion": "4.11.0",
```
Comment (JasBeilin) [inline Packs/CortexXDR/pack_metadata.json:5]: Why minor?

**Review by JasBeilin — CHANGES_REQUESTED** [review]
Nice. Please check my comments.
1. Make sure you understand the meaning of the first_seen_XXX. 
2. Note the changes in the playbook looks like mistake, please  revert the parts you did not intend to change and keep only those you did.
3. Are you using the black or other formatter? When adding code to existing packs we do not use it as it seems like you changes *a lot* of lines that you did not touch.

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 8]
```diff
@@ -3,3183 +3,3281 @@ commonfields:
...
+          name: first_seen_gte
+        - description: |-
+            All the agents that were first seen before {first_seen_lte}.
+            Supported values:
+            1579039377301 (time in milliseconds)
+            "3 days" (relative date)
+            "2019-10-21T23:45:00" (date)
+          name: first_seen_lte
+        - description: |-
+            All the agents that were last seen before {last_seen_gte}.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:988]:
```suggestion
            All the agents that were last seen after  {last_seen_gte}.
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 9]
```diff
@@ -3,3183 +3,3281 @@ commonfields:
...
+          name: last_seen_gte
+        - description: |-
+            All the agents that were last seen before {last_seen_lte}.
+            Supported values:
+            1579039377301 (time in milliseconds)
+            "3 days" (relative date)
+            "2019-10-21T23:45:00" (date)
+          name: last_seen_lte
+        - name: username
+          description: The usernames to query for, accepts a single user, or comma-separated list of usernames.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:1002]:
```suggestion
          description: The usernames to query for. Accepts a single user, or comma-separated list of usernames.
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 10]
```diff
@@ -3,3183 +3,3281 @@ commonfields:
...
+          isArray: true
+        - name: new_alias_name
+          required: true
+          description: The requested alias name to change to.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:1006]:
```suggestion
          description: The alias name to change to.
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/README.md [hunk 11]
```diff
@@ -3187,3 +3187,42 @@ To set up the mirroring:
...
+`xdr-endpoint-alias-change`
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| status | The status of the endpoint to filter. Possible values are: connected, disconnected. | Optional | 
+| endpoint_id_list | A comma-separated list of endpoint IDs. | Optional | 
+| dist_name | A comma-separated list of distribution package names or installation package names.<br/>Example: dist_name1,dist_name2. | Optional | 
+| ip_list | A comma-separated list of IP addresses.<br/>Example: 8.8.8.8,1.1.1.1. | Optional | 
+| group_name | The group name to which the agent belongs.<br/>Example: group_name1,group_name2. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/README.md:3206]:
```suggestion
| group_name | A comma-separated list of group names to which the agent belongs.<br/>Example: group_name1,group_name2. | Optional | 
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/README.md [hunk 12]
```diff
@@ -3187,3 +3187,42 @@ To set up the mirroring:
...
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| status | The status of the endpoint to filter. Possible values are: connected, disconnected. | Optional | 
+| endpoint_id_list | A comma-separated list of endpoint IDs. | Optional | 
+| dist_name | A comma-separated list of distribution package names or installation package names.<br/>Example: dist_name1,dist_name2. | Optional | 
+| ip_list | A comma-separated list of IP addresses.<br/>Example: 8.8.8.8,1.1.1.1. | Optional | 
+| group_name | The group name to which the agent belongs.<br/>Example: group_name1,group_name2. | Optional | 
+| platform | The endpoint platform. Valid values are\: "windows", "linux", "macos", or "android". . Possible values are: windows, linux, macos, android. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/README.md:3207]:
```suggestion
| platform | A comma-separated list of endpoint platforms. Possible values are: windows, linux, macos, android. | Optional | 
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/README.md [hunk 13]
```diff
@@ -3187,3 +3187,42 @@ To set up the mirroring:
...
+| --- | --- | --- |
+| status | The status of the endpoint to filter. Possible values are: connected, disconnected. | Optional | 
+| endpoint_id_list | A comma-separated list of endpoint IDs. | Optional | 
+| dist_name | A comma-separated list of distribution package names or installation package names.<br/>Example: dist_name1,dist_name2. | Optional | 
+| ip_list | A comma-separated list of IP addresses.<br/>Example: 8.8.8.8,1.1.1.1. | Optional | 
+| group_name | The group name to which the agent belongs.<br/>Example: group_name1,group_name2. | Optional | 
+| platform | The endpoint platform. Valid values are\: "windows", "linux", "macos", or "android". . Possible values are: windows, linux, macos, android. | Optional | 
+| alias_name | A comma-separated list of alias names.<br/>Examples: alias_name1,alias_name2. | Optional | 
+| isolate | Specifies whether the endpoint was isolated or unisolated. Possible values are: isolated, unisolated.  Note: This argument returns only the first endpoint that matches. | Optional | 
+| hostname | Hostname<br/>Example: hostname1,hostname2. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/README.md:3210]:
```suggestion
| hostname | A comma-separated list of hostnames.<br/>Example: hostname1,hostname2. | Optional | 
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/README.md [hunk 14]
```diff
@@ -3187,3 +3187,42 @@ To set up the mirroring:
...
+| alias_name | A comma-separated list of alias names.<br/>Examples: alias_name1,alias_name2. | Optional | 
+| isolate | Specifies whether the endpoint was isolated or unisolated. Possible values are: isolated, unisolated.  Note: This argument returns only the first endpoint that matches. | Optional | 
+| hostname | Hostname<br/>Example: hostname1,hostname2. | Optional | 
+| first_seen_gte | All the agents that were first seen after {first_seen_gte}.<br/>Supported values:<br/>1579039377301 (time in milliseconds)<br/>"3 days" (relative date)<br/>"2019-10-21T23:45:00" (date). | Optional | 
+| first_seen_lte | All the agents that were first seen before {first_seen_lte}.<br/>Supported values:<br/>1579039377301 (time in milliseconds)<br/>"3 days" (relative date)<br/>"2019-10-21T23:45:00" (date). | Optional | 
+| last_seen_gte | All the agents that were last seen before {last_seen_gte}.<br/>Supported values:<br/>1579039377301 (time in milliseconds)<br/>"3 days" (relative date)<br/>"2019-10-21T23:45:00" (date). | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/README.md:3213]:
```suggestion
| last_seen_gte | All the agents that were last seen after {last_seen_gte}.<br/>Supported values:<br/>1579039377301 (time in milliseconds)<br/>"3 days" (relative date)<br/>"2019-10-21T23:45:00" (date). | Optional | 
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/README.md [hunk 15]
```diff
@@ -3187,3 +3187,42 @@ To set up the mirroring:
...
+| hostname | Hostname<br/>Example: hostname1,hostname2. | Optional | 
+| first_seen_gte | All the agents that were first seen after {first_seen_gte}.<br/>Supported values:<br/>1579039377301 (time in milliseconds)<br/>"3 days" (relative date)<br/>"2019-10-21T23:45:00" (date). | Optional | 
+| first_seen_lte | All the agents that were first seen before {first_seen_lte}.<br/>Supported values:<br/>1579039377301 (time in milliseconds)<br/>"3 days" (relative date)<br/>"2019-10-21T23:45:00" (date). | Optional | 
+| last_seen_gte | All the agents that were last seen before {last_seen_gte}.<br/>Supported values:<br/>1579039377301 (time in milliseconds)<br/>"3 days" (relative date)<br/>"2019-10-21T23:45:00" (date). | Optional | 
+| last_seen_lte | All the agents that were last seen before {last_seen_lte}.<br/>Supported values:<br/>1579039377301 (time in milliseconds)<br/>"3 days" (relative date)<br/>"2019-10-21T23:45:00" (date). | Optional | 
+| username | The usernames to query for, accepts a single user, or comma-separated list of usernames. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/README.md:3215]:
```suggestion
| username | The usernames to query for. Accepts a single user, or comma-separated list of usernames. | Optional | 
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/README.md [hunk 16]
```diff
@@ -3187,3 +3187,42 @@ To set up the mirroring:
 
 Newly fetched incidents will be mirrored in the chosen direction. However, this selection does not affect existing incidents.
 **Important Note:** To ensure the mirroring works as expected, mappers are required, both for incoming and outgoing, to map the expected fields in Cortex XSOAR and Palo Alto Networks Cortex XDR - Investigation and Response.
+### xdr-endpoint-alias-change
+***
+Gets a list of endpoints according to the passed filters, and changes their alias name. Filtering by multiple fields will be concatenated using AND condition (OR is not supported).
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/README.md:3192]:
```suggestion
Gets a list of endpoints according to the passed filters, and changes their alias name. Filtering by multiple fields will be concatenated using the AND condition (OR is not supported).
```

File: Packs/CortexXDR/ReleaseNotes/4_11_0.md [hunk 17]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Palo Alto Networks Cortex XDR - Investigation and Response Dev
+- Added a new command, ***xdr-endpoint-alias-change***.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/ReleaseNotes/4_11_0.md:4]:
```suggestion
- Added  the ***xdr-endpoint-alias-change*** command.
```

**Comment by ShirleyDenkberg** [comment]
@JasBeilin Doc review completed.

**Review by JasBeilin — CHANGES_REQUESTED** [review]
Make sure to check the flow Binat used.

File: Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule_test.py [hunk 18]
```diff
@@ -3058,3 +3058,20 @@ def test_add_or_remove_tag_endpoint_command(requests_mock, args, expected_filter
...
+    when:
+    - executing the endpoint-alias-change command
+    then:
+    - make sure the body request was sent as expected to the api.
+    """
+    client = CoreClient(base_url=f'{Core_URL}/public_api/v1/', headers={})
+    mocker_set = mocker.patch.object(client, 'set_endpoints_alias')
+    from CoreIRApiModule import endpoint_alias_change_command
+    endpoint_alias_change_command(client=client, status="connected", new_alias_name='test')
+    assert mocker_set.call_args[1] == {'filters': [{'field': 'endpoint_status',
```
Comment (JasBeilin) [inline Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule_test.py:3076]: What happens when no filters were provided? lets add this usecase to the test.

**Review by JasBeilin — COMMENTED** [review]
Please add gracefully handling when no filters are provided.

File: Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule.py [hunk 19]
```diff
@@ -1769,6 +1789,58 @@ def get_endpoints_command(client, args):
...
+
+    # create filters
+    filters: list[dict[str, str]] = create_request_filters(
+        status=status, username=username_list, endpoint_id_list=endpoint_id_list, dist_name=dist_name_list,
+        ip_list=ip_list, group_name=group_name_list, platform=platform_list, alias_name=alias_name_list, isolate=isolate,
+        hostname=hostname_list, first_seen_gte=first_seen_gte, first_seen_lte=first_seen_lte,
+        last_seen_gte=last_seen_gte, last_seen_lte=last_seen_lte, scan_status=scan_status
+    )
+    if not filters:
+        raise DemistoException('No filters were given')
```
Comment (JasBeilin) [inline Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule.py:1835]:
```suggestion
        raise DemistoException('Please provide at least one filter.')
```

File: Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule_test.py [hunk 20]
```diff
@@ -3058,3 +3058,37 @@ def test_add_or_remove_tag_endpoint_command(requests_mock, args, expected_filter
...
+    - executing the endpoint-alias-change command
+    then:
+    - make sure the correct error message wil raise.
+    """
+    client = CoreClient(base_url=f'{Core_URL}/public_api/v1/', headers={})
+    mocker.patch.object(client, 'set_endpoints_alias')
+    from CoreIRApiModule import endpoint_alias_change_command
+    with pytest.raises(Exception) as e:
+        endpoint_alias_change_command(client=client, new_alias_name='test')
+    assert e.value.message == 'No filters were given'
```
Comment (JasBeilin) [inline Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule_test.py:3094]:
```suggestion
    assert e.value.message == 'Please provide at least one filter.'
```

File: Packs/Core/ReleaseNotes/1_3_31.md [hunk 21]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+##### Investigation & Response
+- Updated the Docker image to: *demisto/python3:3.10.10.48392*.
+Added the ***core-endpoint-alias-change*** command that gets a list of endpoints according to the passed filters, and changes their alias name.
```
Comment (JasBeilin) [inline Packs/Core/ReleaseNotes/1_3_31.md:5]:
```suggestion
- Added the ***core-endpoint-alias-change*** command that gets a list of endpoints according to the passed filters, and changes their alias name.
```

**Review by JasBeilin — COMMENTED** [review]
Great work. Please see my comments.

File: Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule_test.py [hunk 22]
```diff
@@ -3058,3 +3058,47 @@ def test_add_or_remove_tag_endpoint_command(requests_mock, args, expected_filter
...
+                                  'operator': 'IN', 'value': ['connected']}], 'new_alias_name': ""}
+
+
+@pytest.mark.parametrize('input, expected_output', [("test", excepted_output_1),
+                                                    ('""', excepted_output_2)])
+def test_endpoint_alias_change_command__diffrent_alias_new_names(mocker, input, expected_output):
+    """
+    Given:
+    - command arguments with endpoint filters, in the first case the new alias name is a string of text,
+    and in the second case the new alias name is a string of only double quotes.
```
Comment (JasBeilin) [inline Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule_test.py:3075]: you can just use this format next time, as docstrings are meant to be short and exact:

Given:
    - valid new alias name as string
    - empty new alias name (due to xsoar limitation, represented by a string of double quote)
    
   Then:
       Makes sure the request body is created correctly.

**Review by JasBeilin — APPROVED** [review]
Great job!

**Pull request merged**